### PR TITLE
[GSoC] Show Avg Interval in Notes Mode (follow up to #12035)

### PR DESCRIPTION
**Pull request by viciousAegis**
## Purpose / Description
Continuation of #12035. Adds functionality to see average interval of note in notes mode of browser.

## Approach
Create a function to average the intervals of non-new and non-learning cards

## How Has This Been Tested?

Unit Test added and tested on device.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt [hunk 1]
```diff
@@ -462,6 +463,8 @@ open class Card : Cloneable {
 
     fun avgEaseOfNote() = avgEase(note())
 
+    fun avgIntervalOfNote() = avgInterval(note())
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt:466]: We shouldn't have dependencies from Card -> `NoteService`

Define these as extensions on `Card` within the `NoteService` file

File: AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/NoteService.kt [hunk 2]
```diff
@@ -211,6 +211,12 @@ object NoteService {
         return nonNewCards.average { it.factor }?.let { it / 10 }?.toInt()
     }
 
+    fun avgInterval(note: Note): Int? {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/NoteService.kt:214]: quick 'javadoc' explaining what a null return means

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 3]
```diff
@@ -2364,6 +2360,24 @@ open class CardBrowser :
...
 
+        private fun getIntervalForCards(): String {
+            return when (card.type) {
+                Consts.CARD_TYPE_NEW -> AnkiDroidApp.instance.getString(R.string.card_browser_interval_new_card)
+                Consts.CARD_TYPE_LRN -> AnkiDroidApp.instance.getString(R.string.card_browser_interval_learning_card)
+                else -> Utils.roundedTimeSpanUnformatted(AnkiDroidApp.instance, card.ivl * Stats.SECONDS_PER_DAY)
+            }
+        }
+
+        private fun getAvgIntervalForNotes(): String {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:2371]: I'd call this `getAvgIntervalForNotes` - work occurs, and `get` implies that it's a quick operation

**Review by david-allison — CHANGES_REQUESTED** [review]
One structural change, rest looks good

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt (hunk 1 above)
Comment (viciousAegis) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt:466]: What is the format for doing that?

**Review by david-allison — APPROVED** [review]
Beauty

**Comment by viciousAegis** [comment]
Needs #12400 to merge first

**Comment by viciousAegis** [comment]
Ready 👍

**Review by Arthur-Milchior — APPROVED** [review]

**Comment by david-allison** [comment]
This is in `main`, this PR needs to be rebased once `main` is green again

**Comment by Arthur-Milchior** [comment]
I resolved the conflict

File: AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/NoteService.kt [hunk 4]
```diff
@@ -217,6 +217,16 @@ object NoteService {
 
     fun totalReviews(note: Note) = note.cards().sumOf { it.reps }
 
+    /**
+     * Returns the average interval of all the non-new and non-learning cards in the note,
+     * or if all the cards in the note are new or learning, returns null
+     */
+    fun avgInterval(note: Note): Int? {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/NoteService.kt:224]: Personally, I'd even go with 
```kotlin
    fun avgInterval(note: Note) = 
        note.
        cards().
        filter { it.type != Consts.CARD_TYPE_NEW && it.type != Consts.CARD_TYPE_LRN }.
        average { it.ivl }?.
        toInt()
```
or even
```Kotlin
    fun Note.avgInterval() =
        cards().
        filter { it.type != Consts.CARD_TYPE_NEW && it.type != Consts.CARD_TYPE_LRN }.
        average { it.ivl }?.
        toInt()
```

**Review by Arthur-Milchior — APPROVED** [review]

**Pull request merged**